pkg/providers/tinkerbell/assert.go

```diff
@@ -294,7 +283,7 @@ func (v *ValidatableTinkerbellCAPI) WorkerNodeHardwareGroups() []WorkerNodeHardw
 }
 
 // HardwareSelector returns the HardwareSelector of the first LabelSelector in TinkerbellMachineTemplate hardware affinity.
-func HardwareSelector(machineTemplate v1beta1.TinkerbellMachineTemplate) v1alpha1.HardwareSelector {
+func HardwareSelector(machineTemplate tinkerbellv1.TinkerbellMachineTemplate) v1alpha1.HardwareSelector {
```

Comment:
Thanks for restructuring and taking out some of the unused references for AssertionsForScaleUpDown. This should have been in my PR, but could you also remove the HardwareSelector() function + tests. It should no longer be used with the current assertion